src/transformers/models/grounding_dino/configuration_grounding_dino.py

```diff
@@ -0,0 +1,378 @@
+# coding=utf-8
+# Copyright 2023 IDEA Research and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Grounding DINO model configuration"""
+import os
+from typing import Union
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+from ..auto import CONFIG_MAPPING
+
+
+logger = logging.get_logger(__name__)
+
+GROUNDING_DINO_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "EduardoPacheco/grounding-dino-tiny": "https://huggingface.co/EduardoPacheco/grounding-dino-tiny/resolve/main/config.json",
+}
+
+
+class GroundingDINOTextPrenetConfig(PretrainedConfig):
```

Comment:
why not just the textConfig? And try to follow our multimodal models!

Reply:
What `textConfig`? I copied the structure from `CLIP`'s implementation

Reply:
What I mean by that is that this should just be the `GroundingDinoTextConfig` there is no need for `Prenet` no?

Reply:
Ohh got it, yeah  I believe we could remove the `Prenet` part. Previously, I added the `Prenet` because I saw in the `make fixup` `prenet`, `decoder`, `encoder`, and `backbone` were ways to ignore the class, but there's no reason why we would want that anyway

Reply:
https://github.com/huggingface/transformers/pull/26087/commits/4553ad1695cc41c6a613c8272c7faa1462f1988d